cpp/src/arrow/compute/util.cc

```diff
@@ -58,6 +58,13 @@ void TempVectorStack::release(int id, uint32_t num_bytes) {
   --num_vectors_;
 }
 
+void TempVectorStack::CheckAllocSizeValid(int64_t estimate_alloc_size) {
```

Comment:
Could you return `arrow::Status` instead of `void` here?

`TempVectorStack::alloc()` will not be able to use it for now but `Hashing32::HashBatch()` can use it.

Reply:
Yes, reasonable. Refactored!